src/coreclr/jit/simd.cpp

```diff
@@ -910,6 +910,30 @@ CorInfoType Compiler::getBaseJitTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeH
     if (simdBaseJitType != CORINFO_TYPE_UNDEF)
     {
         setUsesSIMDTypes(true);
+
+        CORINFO_CLASS_HANDLE* pCanonicalHnd = nullptr;
+        switch (size)
+        {
+            case 8:
+                pCanonicalHnd = &m_simdHandleCache->CanonicalSimd8Handle;
+                break;
+            case 12:
+                // There is no need for a canonical SIMD12 handle because it is always Vector3.
+                break;
+            case 16:
+                pCanonicalHnd = &m_simdHandleCache->CanonicalSimd16Handle;
+                break;
+            case 32:
+                pCanonicalHnd = &m_simdHandleCache->CanonicalSimd32Handle;
+                break;
+            default:
+                unreached();
+        }
+
+        if ((pCanonicalHnd != nullptr) && (*pCanonicalHnd == NO_CLASS_HANDLE))
+        {
+            *pCanonicalHnd = typeHnd;
+        }
```

Comment:
So essentially the first SIMD handle we see in this function becomes the canonical handle for this compilation?

Reply:
Yep. It's a workaround of course, but doing this "properly" would imply breaking the assumption that (most / all that are looked at) `varTypeIsStruct` locals have handles. Which is work that will be done at some point, but not now.